src/content/reference/react/captureOwnerStack.md

```diff
@@ -0,0 +1,301 @@
+---
+title: captureOwnerStack
+---
+
+<Wip>
+
+**This API is experimental and is not available in a stable version of React yet.**
+
+You can try it by upgrading React packages to the most recent experimental version:
+
+- `react@experimental`
+
+Experimental versions of React may contain bugs. Don't use them in production.
+
+</Wip>
+
+<Intro>
+
+`captureOwnerStack` reads the current Owner Stack in development and returns it as a string if available.
+
+```js
+const stack = captureOwnerStack();
+```
+
+</Intro>
+
+<InlineToc />
+
+---
+
+## Reference {/*reference*/}
+
+### `captureOwnerStack()` {/*captureownerstack*/}
+
+Call `captureOwnerStack` to get the current Owner Stack.
+
+```js {5,5}
+import {captureOwnerStack} from 'react';
+
+function Component() {
+  if (process.env.NODE_ENV !== 'production') {
+    const ownerStack = captureOwnerStack();
+    console.log(ownerStack);
+  }
+}
+```
+
+#### Parameters {/*parameters*/}
+
+`captureOwnerStack` does not take any parameters.
+
+#### Returns {/*returns*/}
+
+`captureOwnerStack` returns `string | null`.
+
+If no Owner Stack is available, it returns an empty string.
+Outside of development builds, `null` is returned.
+
+#### Caveats {/*caveats*/}
+
+- Owner Stacks are only available in development. `captureOwnerStack` will always return `null` outside of development.
+
+<DeepDive>
+
+#### Owner Stack vs Component Stack {/*owner-stack-vs-component-stack*/}
+
+The Owner Stack is different from the Component Stack available in React error handlers like [`errorInfo.componentStack` in `onUncaughtError`](/reference/react-dom/client/hydrateRoot#show-a-dialog-for-uncaught-errors).
+
+For example, consider the following code:
+
+<Sandpack>
+
+```js src/App.js
+import {Suspense} from 'react';
+
+function SubComponent({disabled}) {
+  if (disabled) {
+    throw new Error('disabled');
+  }
+}
+
+export function Component({label}) {
+  return (
+    <fieldset>
+      <legend>{label}</legend>
+      <SubComponent key={label} disabled={label === 'disabled'} />
+    </fieldset>
+  );
+}
+
+function Navigation() {
+  return null;
+}
+
+export default function App({children}) {
+  return (
+    <Suspense fallback="loading...">
+      <main>
+        <Navigation />
+        {children}
+      </main>
+    </Suspense>
+  );
+}
+```
+
+```js src/index.js
+import {captureOwnerStack} from 'react';
+import {createRoot} from 'react-dom/client';
+import App, {Component} from './App.js';
+import './styles.css';
+
+createRoot(document.createElement('div'), {
+  onUncaughtError: (error, errorInfo) => {
+    // The stacks are logged instead of showing them in the UI directly to highlight that browsers will apply sourcemaps to the logged stacks.
+    // Note that sourcemapping is only applied in the real browser console not in the fake one displayed on this page.
```

Comment:
Would shorten the line length of this comment so you don't force a hscroll on desktop at least:

<img width="868" alt="Screenshot 2025-01-19 at 10 42 53 AM" src="https://github.com/user-attachments/assets/5f2e467f-cd85-45a1-8cd7-2be4f0d6abd1" />

Reply:
done